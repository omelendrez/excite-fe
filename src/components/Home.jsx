import React from "react";
import { Layout, Spin } from "antd";
import { useSelector } from "react-redux";
import "./home.css";

const Home = (props) => {
  const { loading } = useSelector((state) => state.wakeUp);

  return (
    <>
      {loading && (
        <div className="spinner">
          <Spin size="large" />
        </div>
      )}
      <Layout>{!loading && <div className="home" />}</Layout>
    </>
  );
};

export default Home;
