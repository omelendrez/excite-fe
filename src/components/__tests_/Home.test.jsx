// src/test/Home.test.jsx
import React from "react";
// Replace this with the appropriate location of your component
import Home from "../Home";
// Replace this with the appropriate location of your testing utility
import renderConnected from "../../utils/test/renderConnected";

describe("<Home/>", () => {
  let wrapper, getByText;
  const initialState = {
    // ... Add your initial testing state here
  };

  beforeEach(() => {
    const utils = renderConnected(<Home />, { initialState });
    wrapper = utils.container;
    getByText = utils.getByText;
  });

  it("renders the component", () => {
    expect(wrapper.querySelector(".home")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
