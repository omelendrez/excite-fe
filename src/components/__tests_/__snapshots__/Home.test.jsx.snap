// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="ant-layout"
>
  <div
    className="home"
  />
</section>
`;
